hydromt/workflows/mesh.py

```diff
@@ -84,38 +88,65 @@ def create_mesh2d_from_region(
     kind = next(iter(region))
     if kind == "mesh":
         uds = parse_region_mesh(region)
-        return _create_mesh2d_from_mesh(
+        return create_mesh2d_from_mesh(
             uds,
             grid_name=region.get("grid_name", None),
             logger=logger,
             crs=crs,
             bounds=region.get("bounds", None),
         )
 
-    if kind == "bbox":
+    if kind == "bbox" or kind == "geom":
         if not res:
-            raise ValueError("res argument required for kind bbox")
-        geom = parse_region_bbox(region, crs=crs)
-        return _create_mesh2d_from_bbox(geom, res=res, align=align)
-
-    if kind == "geom":
-        if not res:
-            raise ValueError("res argument required for kind geom")
-        data_catalog = data_catalog or DataCatalog()
-        geom = parse_region_geom(region, crs=crs, data_catalog=data_catalog)
-        return _create_mesh2d_from_geom(geom, res=res, align=align)
+            raise ValueError(f"res argument required for kind {kind}")
+        if kind == "bbox":
+            geom = parse_region_bbox(region, crs=region_crs)
+            clip_to_geom = False
+        else:
+            geom = parse_region_geom(region, crs=region_crs, data_catalog=data_catalog)
+            clip_to_geom = True
+        if crs is not None:
+            crs = utils.parse_crs(crs, bbox=geom.total_bounds)
+            geom = geom.to_crs(crs)
+        return create_mesh2d_from_geom(
+            geom, res=res, align=align, clip_to_geom=clip_to_geom
+        )
```

Comment:
This is way harder to read than what I had.

In that case I would try to go for 

```py
if kind in ["bbox", "geom"]:
  if not res:
    raise

  geom = parse_region_bbox() if kind == "bbox" else parse_region_geom()
  clip_to_geom = kind == "geom"

  if crs is not None:
    crs = ...
    geom = geom.to_crs(crs)

  return create_mesh2d_from_geom()
```

Reply:
Sure that also works. I just am not too found of if and else statement in one line but it works if you prefer it.